File: src/rbpi_blink.c
#include <stdio.h>
#include <stdlib.h>
#include <stdarg.h>
#include <unistd.h>


#include "pr_utils.h" 
#include "cce_1_def.h" 


/*

gcc -o rbpi_blink rbpi_blink.c -lpigpio -lpthread

*/


int main(int argc, char *argv[])
{

 
   printf("Monitoring gpios");
 
	int err= init_cce_1();
	if (err) {
		printf("initialization of hardware failed with err %d \n\r", err);
		return err; 
	}
   
	gpioSetMode(LEDgreen, PI_OUTPUT); // Set LEDgreen== GPIO18 as output.
	
for (;;)
  {
    gpioWrite (LEDgreen, 1) ;	// On
    sleep (1) ;		// mS
    gpioWrite (LEDgreen, 0) ;	// Off
    sleep (1) ;
  }


   gpioTerminate();
}

File: python/rbpi_blink.py
import time

from pr_utils import *  # includes import pigpio1  
from upctrl_pinfunctions import *

def main() :
   print("Control Green LED")
   pi, err = init_cce_1()
   if err or (pi == None ) :
        print(f"initialization of hardware failed with err {err}")
        return err

   pi.set_mode(LEDgreen, pigpio.OUTPUT)  # Set GPIO18 as output.

   try:
      while True:
        print( "Ledd on")
        pi.write(LEDgreen, 1)  # On
        time.sleep(2)  # you need some time to see it
        # remove both sleep statements to see what happens
        print( "Ledd off")
        pi.write(LEDgreen, 0)  # Off
        time.sleep(2)
   except KeyboardInterrupt:
      pass
   finally:
      pi.stop()


if __name__ == "__main__":
    main()
